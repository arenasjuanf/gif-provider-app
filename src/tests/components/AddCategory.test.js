import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";
import '@testing-library/jest-dom'

describe("AddCategory tests", () => {

    const setCategories = jest.fn();
    let wrapper;

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories } />);
    })

    test("the component should display correctly", () => {
        expect(wrapper).toMatchSnapshot();
    })

    test("show change text input", () => {
        const input = wrapper.find("input");
        const value = "chanclas";
        input.simulate('change',{ target: {value} })
        expect( wrapper.find("p").text().trim() ).toBe(value);
    })

    test("it shouldn't do submit", () => {
        
        const submit = wrapper.find("form").simulate("submit", { preventDefault(){} });

        expect(setCategories).not.toHaveBeenCalled();
    })


    test("SetCategories and clear input", () => {
        
        const value = "chanclas";

        const input = wrapper.find("input");
        input.simulate('change',{ target: {value} })
        const form = wrapper.find("form").simulate('submit', { preventDefault(){}});
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect( input.prop('value')).toBe('');

    })

})