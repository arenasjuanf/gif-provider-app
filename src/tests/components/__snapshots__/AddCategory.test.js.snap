// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory tests the component should display correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
